src/bp-notifications/bp-notifications-functions.php

```diff
@@ -471,10 +471,9 @@ function bp_notifications_delete_notifications_from_user( $user_id, $component_n
  * @since 2.5.0
  *
  * @param int $user_id ID of the user who is about to be deleted.
- * @return int|false The number of rows deleted, or false on error.
  */
 function bp_notifications_delete_notifications_on_user_delete( $user_id ) {
-	return BP_Notifications_Notification::delete( array(
+	BP_Notifications_Notification::delete( array(
```

Comment:
This change is making me nervous, my concern is if a plugin is using this function to actually get a returned value it might generate issues with it. Could we include a check to be sure the current action is the hook it's attached to, in other words that it's not being use directly ?

Reply:
Let's create a different function with the side effect instead of bundling those two together since they serve different purposes.

Reply:
Ok, this is what I suggest, it the current action is not corresponding to the hook this function is attached to > throw a doing it wrong.

```suggestion
	if ( 'wpmu_delete_user' !== current_action() ) {
		_doing_it_wrong( __FUNCTION__, __( 'This function is only used once a user is deleted. Please use `bp_notifications_delete_notifications_from_user()` instead.', 'buddypress' ), '12.0.0' );
	}
	
	BP_Notifications_Notification::delete( array(
```

Could you also update the `bp_notifications_delete_notifications_from_user()` function so that its arguments have default values, see below: 

```php
function bp_notifications_delete_notifications_from_user( $user_id = 0, $component_name = false, $component_action = false ) {
```

Reply:
@imath If I wanted to use this function in a custom/new hook, this new `doing_it_wrong` error would stop me.

Reply:
Right! Thanks for your feedback. We're actually directly using this function in `bp_notifications_delete_notifications_on_delete_user()`. Here's what I would suggest for now. Let's leave this part the way it is for now, commit the rest of the PR, and open a specific ticket about it to find the best way to deal with it.

Reply:
I mean, we are doing, and will do more of, this change in several places of the codebase. Why are you singling this example out for discussion?

We should do just like the other examples, remove the `return` and let those affected adapt to this change. Unless you are planning to discuss every single instance since there has been a lot, and I'm not even done with several components yet, so I'm sure they will be more.

Reply:
Here's how I see things: if I see a potential problem with plugins about a change, I will try to prevent it. That's one less than all the ones I haven't seen. If being in line with WPCS means potentially break a plugin, then I choose not to be in line with WPCS and I simply add `// phpcs: ignore`.

What you say is: there's a potential problem with plugins, but nevermind they will adapt. First I'm pretty sure they wouldn't, and in the end the end users will complain in forums and we'll loose time to understand what's going wrong.

So we can try to find the best way to deal with it, or you can commit, and I will revert this particular change, open a ticket to try to find the best way to deal with it.

BTW, it looks like you committed my suggestion into your PR according to failing tests. Weird.

Reply:
I see some misunderstandings in your feedback, so I'm suggesting moving the conversation about PHPDoc and WPCS to [Slack](https://wordpress.slack.com/archives/C02RQBYUG/p1674399913706989) so that I can clarify what I meant there.

For the actual change, we can keep the function with the same name and create a different one for the `do_action` hook. The name of the function with the returned value is misleading though, `bp_notifications_delete_notifications_on_user_delete`, since it was created to be run in a `do_action`. But this approach will account for your concerns, in case a plugin author is using it.

Reply:
Sure, your solution is fine with me 💯